### PR TITLE
Return the correct error while trying to update the root tenant

### LINKED ISSUE #5396: Internal Server Error when update root tenant
## I'm submitting a ...

-  bug report

## Traffic Control components affected ...

-  Traffic Portal

## Current behavior:

1. Go to TP 
2. Click on User Admin -> Tenants
3. Try to update root tenant. Notice that it will generate Internal Server Error message
## Expected behavior:

~~TP should let user update root tenant~~
When attempting to update the immutable root tenant, the user should get a 400 Bad Request with a message along the lines of "Root tenant is immutable".
## Anything else:

ERROR: api.go:209: 2020-12-29T17:08:32.859350184Z: 96.118.16.179:37360 checking tenant authorized: sql: Scan error on column index 0, name "parent_id": converting NULL to int is unsupported

### PR DESCRIPTION
**Pull request by srijeet0406**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5396  

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Ops

## What is the best way to verify this PR?

Go to traffic portal and try to update the `root` tenant
You will receive a `400 Bad Request` error with the correct error msg.

## If this is a bug fix, what versions of Traffic Control are affected?

- master

## The following criteria are ALL met by this PR

- [x] This PR includes tests 
- [ ] This PR includes documentation 
- [x] This PR includes an update to CHANGELOG.md 
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/apitenant/tenant.go [hunk 1]
```diff
@@ -184,13 +186,16 @@ func (ten *TOTenant) IsTenantAuthorized(user *auth.CurrentUser) (bool, error) {
...
 		var parentID int
 		tx := ten.APIInfo().Tx.Tx
-		err = tx.QueryRow(`SELECT parent_id FROM tenant WHERE id = ` + strconv.Itoa(*ten.ID)).Scan(&parentID)
-		if err != nil {
-			return false, err
-		}
-		if parentID == *ten.ParentID {
-			// parent not being changed
-			return ok, err
+		// If its the rot tenant, dont check for parent
```
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/apitenant/tenant.go:189]: * its
* rot
* dont

File: traffic_ops/traffic_ops_golang/apitenant/tenant.go [hunk 2]
```diff
@@ -202,7 +207,12 @@ func (ten *TOTenant) IsTenantAuthorized(user *auth.CurrentUser) (bool, error) {
 	return tenant.IsResourceAuthorizedToUserTx(*ten.ParentID, user, ten.APIInfo().Tx.Tx)
 }
 
-func (tn *TOTenant) Update(h http.Header) (error, error, int) { return api.GenericUpdate(h, tn) }
+func (tn *TOTenant) Update(h http.Header) (error, error, int) {
+	if tn.Name != nil && *tn.Name == rootName {
+		return errors.New("cannot update the root tenant"), nil, http.StatusBadRequest
```
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/apitenant/tenant.go:212]: 400-level errors should include a description of why the server rejected the request.

**Review by zrhoffman — CHANGES_REQUESTED** [review]

**Comment by zrhoffman** [comment]
> This PR includes documentation

I can't find the documentation

**Comment by srijeet0406** [comment]
> > This PR includes documentation
> 
> I can't find the documentation

Fixed

**Review by zrhoffman — CHANGES_REQUESTED** [review]

**Review by zrhoffman — APPROVED** [review]

**Pull request merged**